bin/start-pgbouncer-as-service

```diff
@@ -3,88 +3,7 @@
 
 set +e
 
-export PGBOUNCER_PROCESS_REGEX="[v]endor/pgbouncer/bin/pgbouncer"
-export PGBOUNCER_BUILD_PACK_PID
+# shellcheck source=./../.profile.d/pgbouncer.sh
+source "$(dirname "${BASH_SOURCE[0]}")/../.profile.d/pgbouncer.sh"
 
-function inf() {
-  echo -e "INFO:  $* "
-}
-
-function err() {
-  echo -e "ERROR: $* "
-}
-
-function is-enabled() {
-  (
-    shopt -s extglob nocasematch
-    [[ $1 == @(1|true|yes|on) ]]
-  )
-}
-
-function pgbouncer-current-pid() {
-  /bin/ps -eopid,args | grep -E "${PGBOUNCER_PROCESS_REGEX}" | awk '{print $1}'
-}
-
-function is-pgbouncer-running() {
-  local current_pid
-  current_pid=$(pgbouncer-current-pid)
-
-  [[ -n ${current_pid} && ${current_pid} -gt 0 ]] && {
-    inf "pgbouncer-is-running: pid=${current_pid}"
-    return 0
-  }
-
-  inf "pgbouncer-is-not-running"
-  return 1
-}
-
-function is-pgbouncer-service-enabled() {
-  if ! is-enabled "${PGBOUNCER_ENABLED}"; then
-    inf "pgbouncer-is-disabled"
-    return 1
-  fi
-  return 0
-}
-
-function main() {
-  # Do not start the service if PgBouncer is not enabled
-  is-pgbouncer-service-enabled || return 1
-
-  local stderr
-  stderr="$(mktemp)"
-
-  trap '' SIGTERM
-
-  [[ -x bin/start-pgbouncer ]] && bin/start-pgbouncer /bin/bash -c "while true; do sleep 10000; done" 2>"${stderr}" &
-
-  # Translate SIGHUP to the appropriate signal to stop the child (anything
-  # except SIGTERM which is ignored). This *will* cancel the wait and may
-  # lead to the outer subshell exiting before the aux process
-  trap "kill -SIGINT $!" SIGHUP
-
-  sleep 1
-
-  # PgBouncer should be running now; if it isn't then return early
-  is-pgbouncer-running || return 1
-
-  PGBOUNCER_BUILD_PACK_PID=$(pgbouncer-current-pid)
-  export PGBOUNCER_BUILD_PACK_PID
-
-  if [[ -n ${PGBOUNCER_BUILD_PACK_PID} && ${PGBOUNCER_BUILD_PACK_PID} -gt 0 ]]; then
-    set +e
-    mkdir -p tmp/pids || true
-    echo "${PGBOUNCER_BUILD_PACK_PID}" > tmp/pids/pgbouncer.pid
-    inf "pgbouncer-started-ok-pid-s${PGBOUNCER_BUILD_PACK_PID}"
-    return 0
-  else
-    err "pgbouncer-failed-to-start"
-    [[ -s ${stderr} ]] && {
-      err "pgbouncer-stderr:"
-      cat "${stderr}"
-      rm -f "${stderr}"
-    }
-    return 1
-  fi
-}
-
-[[ "$0" != "${BASH_SOURCE[0]}" ]] || main "$@"
+pgbouncer-start-as-a-service
```

Comment:
[nit] Does it make sense to define the function in the shared lib when it's only going to be called here?